src/core/border_router/routing_manager.hpp

```diff
@@ -0,0 +1,259 @@
+     *
+     */
+    Ip6::Prefix EvaluateOmrPrefix(void);
+
+    /**
+     * This method evaluates the on-link prefix for the infra link.
+     *
+     * @returns  The new on-link prefix should be advertised. An invalid on-link prefix
+     *           means we should no longer advertise an on-link prefix on the infra link.
+     *
+     */
+    Ip6::Prefix EvaluateOnLinkPrefix(void);
+
+    /**
+     * This method publishes an OMR prefix in Thread network.
+     *
+     * @param[in]  aOmrPrefix  An OMR prefix.
+     *
+     */
+    void PublishOmrPrefix(const Ip6::Prefix &aOmrPrefix);
+
+    /**
+     * This method unpublishes an OMR prefix if we have done that.
+     *
+     * @param[in]  aOmrPrefix  An OMR prefix.
+     *
+     */
+    void UnpublishOmrPrefix(const Ip6::Prefix &aOmrPrefix);
+
+    /**
+     * This method sends Router Solicitation messages to discovery on-link
+     * prefix on infra links.
+     *
+     * @see HandleRouterAdvertisement
+     *
+     */
+    void SendRouterSolicit(void);
+
+    /**
+     * This method sends Router Advertisement messages to advertise
+     * on-link prefix and route for OMR prefix.
+     *
+     * @param[in]  aNewOmrPrefix  The new OMR prefix to be advertised.
+     * @param[in]  aOnLinkPrefix  The new on-link prefix to be advertised.
+     *
+     */
+    void SendRouterAdvertisement(const Ip6::Prefix &aNewOmrPrefix, const Ip6::Prefix &aNewOnLinkPrefix);
+
+    static void HandleRouterAdvertisementTimer(Timer &aTimer);
+    void        HandleRouterAdvertisementTimer(void);
+
+    static void HandleRouterSolicitTimer(Timer &aTimer);
+    void        HandleRouterSolicitTimer(void);
+
+    static void HandleDiscoveredOnLinkPrefixInvalidTimer(Timer &aTimer);
+    void        HandleDiscoveredOnLinkPrefixInvalidTimer(void);
+
+    void HandleRouterSolicit(const Ip6::Address &aSrcAddress, const uint8_t *aBuffer, uint16_t aBufferLength);
+    void HandleRouterAdvertisement(const Ip6::Address &aSrcAddress, const uint8_t *aBuffer, uint16_t aBufferLength);
+    bool HandlePrefixInfoOption(const RouterAdv::PrefixInfoOption &aPio);
+
+    static TimeMilli GetPrefixExpireTime(uint32_t aValidLifetime);
+
+    uint32_t mInfraIfIndex;
+    char     mInfraIfName[kMaxInfraIfNameLength + 1];
+
+    /**
+     * The OMR prefix loaded from local persistent storage.
+     *
+     */
+    Ip6::Prefix mLocalOmrPrefix;
+
+    /**
+     * The OMR prefix selected to be advertised.
+     *
+     */
+    Ip6::Prefix mAdvertisedOmrPrefix;
+
+    /**
+     * The on-link prefix created based on the local OMR prefix.
+     *
+     */
+    Ip6::Prefix mLocalOnLinkPrefix;
+
+    /**
+     * The on-link prefix selected to be advertised.
+     *
+     */
+    Ip6::Prefix mAdvertisedOnLinkPrefix;
+
+    /**
+     * The on-link prefix we discvered on the infra link by
+     * sending Router Solicitations.
+     *
+     */
+    Ip6::Prefix mDiscoveredOnLinkPrefix;
+
+    TimerMilli mRouterAdvertisementTimer;
```

Comment:
We have timers that ticks in unit of milliseconds, so it will be a bit weird that some of the timers are managed by the `TimeTicker` while others are not. Considering the border router is usually enabled on a relatively powerful platform, I think it is not critical to optimize for memory.